misk/src/main/kotlin/misk/digester/SlidingWindowDigest.kt

```diff
@@ -0,0 +1,182 @@
+package misk.digester
+
+import java.time.Clock
+import java.time.ZonedDateTime
+
+/** WindowDigest holds a t-digest whose data points are scoped to a specific time window. */
+data class WindowDigest<T : TDigest<T>>(
+  val window: Window,
+  val Digest: T
+)
+
+/** Snapshot is the state of a SlidingWindowDigest at a point in time. */
+data class Snapshot(
+  val quantileVals: DoubleArray,  // Values of specific quantiles.
```

Comment:
The compiler gives a warning about equality/hashing not working correctly with arrays. This is because in the JVM `arrayX == arrayY` does not compare elements.

`List<Double>` is safe, though.